zadanie3/zadanie3.cpp

```diff
@@ -84,33 +86,68 @@ bool usun_element(Lista* l, Element* usun)
 	return false;
 }
 
-void swap(Element* a,Element* b)
-{
-    Element tmp;
-    Element *tmpe;
-    tmp = *a;
-    *a = *b;
-    *b = tmp;
-
-    tmpe = a->next;
-    a->next = b->next;
-    b->next = tmpe;
-}
-
 void sort(Lista* l)
 {
-    sort(l->glowa);
+    sort(&(l->glowa));
 }
 
-void sort(Element* head)
+
+/**
+ * W przeciwieństwie do poprzedniej wersji, zamiast zamieniac wskazniki,
+ * zamieniamy wskazniki do wskaznikow
+ * metoda 'bubble' : http://www.sorting-algorithms.com/
+ */
```

Comment:
I tutaj sie zaczyna ciekawie, sam w 100% nie obejmuje tego algorytmu do konca, ten kod to jest copy/paste z malymi zmianami. Pod tym linkiem jest ten algorytm zilustrowany: http://www.sorting-algorithms.com/bubble-sort